File: src/demos/ps/maps/Skure_entrance.java
package demos.ps.maps;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import static demos.ps.oo.PSGame.*;
import demos.ps.oo.City;
import demos.ps.oo.Dungeon;
import demos.ps.oo.PSGame;
import demos.ps.oo.Item;
import demos.ps.oo.PSLibItem.OriginalItem;
import demos.ps.oo.PSMenu;
import demos.ps.oo.PSMenu.EntityClothes;
import demos.ps.oo.PSMenu.EntityType;
import demos.ps.oo.PSMenu.LargeEntity;
import demos.ps.oo.PSMenu.Outcome;
import demos.ps.oo.PSMenu.Scene;
import demos.ps.oo.PSMenu.SpecialEntity;

public class Skure_entrance {


	public static void tunnel() {
		mapswitch(Dungeon.SKURE_TUNNEL_IN);
	}
	
	public static void exit() {
		mapswitch(Planet.DEZORIS, City.SKURE_ENTRANCE.getX(),City.SKURE_ENTRANCE.getY());
	}
	
	public static void spaceship() {
		PSMenu.startScene(Scene.ARTIC, LargeEntity.HAPSBY);
		PSGame.hapsbyRoutine(City.SKURE);
		PSMenu.endScene(Outcome.FADE_HOUSE);
	}
	
	
}

File: src/demos/ps/maps/Skure_entrance.ts
/**
 * Skure_entrance Script
 * TypeScript port of Skure_entrance.java
 */

import { PSGame } from '../PSGame';
import { Planet, City, CityHelper } from '../game/City';
import { Dungeon } from '../game/Dungeon';
import { PSSceneType, LargeEntity, PSMenu } from '../PSMenu';
import { PSOutcome } from '../menu/MenuStack';

export class Skure_entrance {

  public static async tunnel(): Promise<void> {
    await PSGame.mapswitchToDungeon(Dungeon.SKURE_TUNNEL_IN);
  }

  public static exit(): void {
    PSGame.mapswitchToPlanet(Planet.DEZORIS, CityHelper.getX(City.SKURE_ENTRANCE), CityHelper.getY(City.SKURE_ENTRANCE));
  }

  public static async spaceship(): Promise<void> {
    await PSMenu.startSceneWithLargeEntity(PSSceneType.ARTIC, LargeEntity.HAPSBY);
    const dest = await PSGame.hapsbyRoutine(City.SKURE);
    if (dest !== null) {
      // Close the scene onto black first — the travel chain is awaited inline
      // (Java deferred the mapswitch until after endScene)
      await PSMenu.endSceneToBlack();
      await PSGame.spaceshipRoutineStart(City.SKURE, dest);
      return;
    }
    await PSMenu.endSceneWithOutcome(PSOutcome.FADE_HOUSE);
  }
}
